modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/store/withStore.es.js

```diff
@@ -62,16 +63,31 @@ const _handleFieldBlurred = function(properties) {
 			pages: blurredFieldPages
 		});
 	});
+
+	Liferay.fire('ddmFieldBlur', {
+		fieldName: fieldInstance.fieldName,
+		focusDuration: now - (this.fieldFocusDate || now),
+		formId: this.getFormId(),
+		page: this.activePage
+	});
 };
 
 const _handleFieldFocused = function(properties) {
 	const {pages} = this;
 
+	this.fieldFocusDate = new Date();
+
 	handleFieldFocused(pages, properties).then(focusedFieldPages => {
 		this.setState({
 			pages: focusedFieldPages
 		});
 	});
+
+	Liferay.fire('ddmFieldFocus', {
+		fieldName: properties.fieldInstance.fieldName,
```

Comment:
@interaminense, why did you need to use `properties.fieldInstance.fieldName` instead of `fieldInstance.fieldName`?

Reply:
**fieldInstance** is a properties property, but to keep the same pattern as **handleFieldFocused**, I destructured the properties property to access **fieldInstance** directly: https://github.com/natocesarrego/liferay-portal/pull/177/commits/81d8c799b8bb862c78256d8e920b77d409027c91